doc/subsystems/test/ztest.rst

```diff
@@ -52,6 +52,37 @@ src/main.c
    :local:
    :backlinks: top
 
+
+
+A test case project may consist of multiple sub-tests or smaller tests that
+either can be testing functionality or APIs. Functions implementing a test
+should follow the the guidelines below:
+
+* Test cases function names should be prefix with **test_**
+* Test cases should be documented using doxygen
+* Test function names should be unique within the section or component being
+  tested
+
+
+An example can be seen below::
+
+    /**
+     * @brief Test Asserts
+     *
+     * This test verifies asserts
```

Comment:
Since we're providing an example, let's provide a good description.  Maybe something like:

    * This test verifies a positive assertion does not trigger a runtime error.

Reply:
that just complicates things to be honest, this is just to demonstrate the usage in a test, not the test itself. I modified it a bit.